test/gtest/uct/test_device.cc

```diff
@@ -0,0 +1,102 @@
+/**
+* Copyright (c) NVIDIA CORPORATION & AFFILIATES, 2025. ALL RIGHTS RESERVED.
+* See file LICENSE for terms.
+*/
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include <string>
+
+#include <uct/api/uct.h>
+#include "uct_test.h"
+
+extern "C" {
+#include <uct/cuda/base/cuda_iface.h>
+}
+
+#include "cuda/test_kernels.h"
+
+class test_device : public uct_test {
+protected:
+    void init()
+    {
+        CUresult res_drv;
+        CUcontext ctx;
+        int cuda_id;
+        ucs_status_t status;
+
+        uct_test::init();
+
+        cuda_id = std::stoi(
+                GetParam()->dev_name.substr(UCT_DEVICE_CUDA_NAME_LEN));
+        res_drv = cuDeviceGet(&m_cuda_dev, cuda_id);
+        if (res_drv != CUDA_SUCCESS) {
+            ucs_error("cuDeviceGet returned %d.", res_drv);
+            return;
+        }
+
+        status = UCT_CUDADRV_FUNC_LOG_ERR(
+                cuDevicePrimaryCtxRetain(&ctx, m_cuda_dev));
+        if (status != UCS_OK) {
+            return;
+        }
+
+        status = UCT_CUDADRV_FUNC_LOG_ERR(cuCtxPushCurrent(ctx));
+        if (status != UCS_OK) {
+            return;
+        }
+
+        m_receiver = uct_test::create_entity(0);
+        m_entities.push_back(m_receiver);
+
+        m_sender = uct_test::create_entity(0);
+        m_entities.push_back(m_sender);
+
+        m_sender->connect(0, *m_receiver, 0);
+    }
+
+    void cleanup()
+    {
+        (void)UCT_CUDADRV_FUNC_LOG_WARN(cuCtxPopCurrent(NULL));
+        (void)UCT_CUDADRV_FUNC_LOG_WARN(cuDevicePrimaryCtxRelease(m_cuda_dev));
```

Comment:
maybe setting right cuda context to launch kernel should be part of common cuda test infra (can be next pr)